lambdas/data-migration2/src/granulesAndFiles.ts

```diff
@@ -88,6 +96,11 @@ export const migrateGranuleRecord = async (
     granule,
     executionCumulusId
   ));
+
+  if (!cumulusId) {
+    throw new PostgresUpdateFailed(`Granule ${record.granuleId} was not able to be updated in the Postgres table`);
```

Comment:
So this error is thrown because the upsert from `upsertGranuleWithExecutionJoinRecord` affected no rows, but doesn't throw an error?

Reply:
Right. If the `upsert` doesn't return rows, `upsertGranuleWithExecutionJoinRecord` will return an empty array. In that case it made sense to me to have `migrateGranuleRecord` throw an error rather than return undefined. It's caller expects either an error or a valid value (which we could obviously change but this seemed cleaner).

This reminded me to update the docstring:

https://github.com/nasa/cumulus/pull/2162/commits/2e26bd3f46bb0b706434a3d3c4e4eb7bc7b39fb7

Reply:
I might make the content of this message (here and in PDRs) more explicit both for our own benefit and for users

```suggestion
    throw new PostgresUpdateFailed(`Upsert for granule ${record.granuleId} returned no rows. Record was not updated in the Postgres table.`);
```

Reply:
That's much clearer:

https://github.com/nasa/cumulus/pull/2162/commits/ea03d80a16a603b86afc5c4b484f2e56b6072881